### ISSUE TITLE
Unable to use groundhog.library() inside .Rmd file due to groundhog's requirement for authorization via text input

### ISSUE THREAD

**Issue by martinlaw** [CredibilityLab/groundhog#119]
This issue may have similarities with issues #67 and #117 but I believe is distinct.

I'm not sure how to use groundhog in an .Rmd file. When running `groundhog.library` for the first time in the console, it requests authorization to save files, e.g.

`groundhog needs authorization to save files to 'C:/Users/.../R_groundhog/'
Enter 'OK' to provide authorization, and 'NO' not to.`

However, this request does not appear either when knitting an .Rmd file by clicking the "Knit" button, nor when knitting an .Rmd in the console with the command `rmarkdown::render`. When using the "Knit" button, there is simply the error mesage `Execution halted`. When using `rmarkdown::render`, an empty prompt appears, with no text:

![Capture](https://github.com/user-attachments/assets/b48b75da-7b7d-423f-a609-520775154458)

MWE for .Rmd file (with groundhog not installed):

````
---
title: "ghog-test"
output: html_document
---
```{r}
if("groundhog" %in% installed.packages()==FALSE) install.packages("groundhog")
library(groundhog)
groundhog.library("BATSS", date = "2024-07-20")
```
````

How can I write an .Rmd file so that someone who is new to groundhog can successfully knit this file? I imagine that I could include a source file along the lines of 

```
if("rmarkdown" %in% installed.packages()==FALSE) install.packages("rmarkdown")
library(rmarkdown)
rmarkdown::render("test.Rmd") # Type "YES" then hit Enter!!
```

but this seems like a suboptimal solution.

I appreciate from replies in other Issues that you do not use knitr regularly.

**Comment by urisohn** [comment]
Ok, to  make sure i understand, you want to share code that others will knit, and you want to others to knit it without executing any code themselves, just knitting. The requirement to ask for user approval is there to comply with CRAN rules, cannot have groundhog save packages without user approval. 

If this is indeed the situation, you will need to either add some interactivity like the one you are proposing, or you will need to do something I can't, which is to not ask users and just install stuff for them. 

For the interactive solution, why do you not like the one you propose?
For the don't ask don't tell solution, you could install groundhog and also run set.groundhog.folder(<path>) for users and they would not be prompted.

**Comment by martinlaw** [comment]
Yes you understand perfectly. Apologies, I had forgotten that CRAN has such a rule.
 
I am not certain that there is a "Don't Ask, Don't Tell" solution: I amended my code and sent it to a colleague, who also hit an "Execution halted" error. Here is the updated code:

````
---
title: "ghog-test"
output: html_document
---
```{r}
if("groundhog" %in% installed.packages()==FALSE) install.packages("groundhog", repos="https://cloud.r-project.org")
library(groundhog)
if(file.exists("groundhog-folder")==FALSE) dir.create("groundhog-folder")
set.groundhog.folder("groundhog-folder")
groundhog.library("BATSS", date = "2024-07-20")
```
````

It is difficult to test myself as I'm not sure how to repeatedly "remove" groundhog's permission to write and then try again.

For the moment, I think an interactive approach may be necessary -- I simply wanted something as "bulletproof" as possible.

**Comment by urisohn** [comment]
First, to de-authorize groundhog you would delete this folder ('default_path')
` default_path=paste0(path.expand("~"), "/R_groundhog")`
that's where the text file with the authorized path to save to is saved, so if you delete that folder it's as if permission has not been granted.

Second, therefore, if you want to do a don't ask don't tell, you could put this in your script:
default_folder <- paste0(path.expand("~"), "/R_groundhog")
#If it does not exist, set the default folder to it
if (dir.exists(default_folder)==FALSE) {
  message("Creating folder ", default_folder, " to install version controlled packages with {groundhog} to")
  set.groundhog.folder(default_folder)
  
  }

**Comment by martinlaw** [comment]
Thank you. This works perfectly, for both me and my colleague acting as the non-groundhog user. I believe the code in my previous post did not work because I was creating a directory with a name differed from the default directory name. 

This Issue is closed as far as I'm concerned. Thank you again.

**Comment by urisohn** [comment]
cool, anything else comes up just let me know
